### PR TITLE
null check with_from items

### PR DESCRIPTION
**Pull request by sierra-moxon**

### CODE REVIEW

**Comment by kltm** [comment]
https://github.com/geneontology/pipeline/issues/346

**Comment by dustine32** [comment]
@mugitty is going to push an update to this `qc_null_check` branch to get the failing `test_go_rule_16` test to pass.

**Comment by mugitty** [comment]
After discussing with @dustine32, we decided to make additional changes. I have committed additional changes

File: ontobio/io/gpadparser.py
Comment (mugitty) [inline ontobio/io/gpadparser.py:1]: Failure of gorule16 is soft.  Need to ensure assoc.evidence.with_support_from is not None before traversing

File: ontobio/io/qc.py
Comment (mugitty) [inline ontobio/io/qc.py:1]: Gorule16 needs to verify that for annotation with code IC, at least one of the with/from is a go term that is not same as the go term for the annotation

**Review by dustine32 — APPROVED** [review]
@mugitty Looks good to me. The travis tests and `validate.py rule` all pass so merge whenever.

File: tests/test_qc.py
Comment (mugitty) [inline tests/test_qc.py:1]: Added test case for Gorule16 where GPAD parser creates annotation with evidence code ECO that does not have a with/from

**Pull request merged**